### PR TITLE
Make `edpm_ovn` compliant with ansible-lint `production` profile

### PR DESCRIPTION
**Pull request by rebtoor**
- Fixed yaml indentation issues
- Added `changed_when`/`failed_when` in `shell` tasks
- Replaced `ignore_errors: true` with `failed_when: false`
- Added FQCN where needed

### CODE REVIEW

File: roles/edpm_ovn/defaults/main.yml [hunk 1]
```diff
@@ -78,16 +78,19 @@ edpm_ovn_ovs_external_ids:
...
+      join(':')) }}{%- endfor -%}
+    {{ chassis_mac_mappings | join(',') }}"
   ovn-encap-ip: "{{ edpm_ovn_encap_ip }}"
   ovn-encap-type: "{{ edpm_ovn_encap_type }}"
   ovn-match-northd-version: true
   ovn-monitor-all: true
   ovn-openflow-probe-interval: "{{ edpm_ovn_of_probe_interval }}"
-  ovn-remote: "{% set db_addresses = [] %}{% for host in edpm_ovn_dbs %}{{ db_addresses.append([edpm_ovn_protocol, host, edpm_ovn_sb_server_port] | join(':')) }}{% endfor %}{{ db_addresses | join(',') }}"
+  ovn-remote:
+    "{% set db_addresses = [] %}{% for host in edpm_ovn_dbs %}
```
Comment (karelyatin) [inline roles/edpm_ovn/defaults/main.yml:91]: failures in molecule are relavant, need to fix extra space in ovn-remote

Comment (rebtoor) [reply roles/edpm_ovn/defaults/main.yml:91]: indeed, thx @karelyatin !

File: roles/edpm_ovn/tasks/configure.yml [hunk 2]
```diff
@@ -62,20 +62,32 @@
...
+      changed_when: true
+      failed_when: false
+
     - name: Add OVS Manager if not exists
       ansible.builtin.shell: >
         ovs-vsctl --timeout=5 --id=@manager -- create Manager target=\"ptcp:6640:127.0.0.1\" -- add Open_vSwitch . manager_options @manager
       when: ovs_manager_configured.rc == 1
+      register: ovs_manager_if_not_exists
+      changed_when: ovs_manager_if_not_exists.rc == 0
+      failed_when: ovs_manager_if_not_exists.rc != 0
```
Comment (fao89) [inline roles/edpm_ovn/tasks/configure.yml:93]: Adoption CI started to fail after this change
https://github.com/openstack-k8s-operators/data-plane-adoption/pull/178#issuecomment-1758200099

Comment (karelyatin) [reply roles/edpm_ovn/tasks/configure.yml:93]: @fao89 https://github.com/openstack-k8s-operators/edpm-ansible/pull/440 fixes that issue. So once it's available in the jobs, issue shouldn't be seen

**Pull request merged**